superduperdb/base/serializable.py

```diff
@@ -161,13 +137,67 @@ class Serializable:
     serialize = _serialize
 
     @property
-    def variables(self) -> t.List[Variable]:
-        return _find_variables(self.dict())
-
-    def set_variables(self, db) -> 'Serializable':
-        r = self.serialize()  # get serialized version of class instance
-        r = _replace_variables(r, db)  # replace variables with values
+    def variables(self) -> t.List['Variable']:
+        out = {}
+        v = _find_variables(self.dict())
+        for var in v:
+            out[var.value] = var
+        return sorted(list(out.values()), key=lambda x: x.value)
+
+    def set_variables(self, db, **kwargs) -> 'Serializable':
+        r = self.serialize(
+            serialize_variables=False
+        )  # get serialized version of class instance
+        r = _replace_variables(r, db, **kwargs)  # replace variables with values
         return self.deserialize(r)  # rebuild new class instance
 
     def dict(self) -> t.Dict[str, t.Any]:
         return asdict(self)
+
+
+class Variable:
```

Comment:
This class allows queries to be included in a model, 
and "crystallized" at runtime.